integrationTests/peerConnections/peerConnections_test.go

```diff
@@ -0,0 +1,192 @@
+package peerConnections
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-crypto-go/signing"
+	"github.com/multiversx/mx-chain-crypto-go/signing/secp256k1"
+	"github.com/multiversx/mx-chain-crypto-go/signing/secp256k1/singlesig"
+	p2p "github.com/multiversx/mx-chain-p2p-go"
+	"github.com/multiversx/mx-chain-p2p-go/config"
+	"github.com/multiversx/mx-chain-p2p-go/libp2p"
+	"github.com/multiversx/mx-chain-p2p-go/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var keyGen = signing.NewKeyGenerator(secp256k1.NewSecp256k1())
+
+func createBaseArgs() libp2p.ArgsNetworkMessenger {
+	prvKey, _ := keyGen.GeneratePair()
+
+	return libp2p.ArgsNetworkMessenger{
+		Marshalizer: &mock.MarshallerMock{},
+		P2pConfig: config.P2PConfig{
+			Node: config.NodeConfig{
+				Port: "0", // auto-select port
+			},
+			KadDhtPeerDiscovery: config.KadDhtPeerDiscoveryConfig{
+				Enabled: false,
+			},
+			Sharding: config.ShardingConfig{
+				Type: p2p.NilListSharder,
+			},
+		},
+		SyncTimer:             &mock.SyncTimerStub{},
+		PreferredPeersHolder:  &mock.PeersHolderStub{},
+		NodeOperationMode:     p2p.NormalOperation,
+		PeersRatingHandler:    &mock.PeersRatingHandlerStub{},
+		ConnectionWatcherType: p2p.ConnectionWatcherTypePrint,
+		P2pPrivateKey:         prvKey,
+		P2pSingleSigner:       &singlesig.Secp256k1Signer{},
+		P2pKeyGenerator:       keyGen,
+	}
+}
+
+func TestPeerConnectionsOnAllSupportedProtocolsShouldExchangeData(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	messengers := make([]p2p.Messenger, 0)
+
+	seederArgs := createBaseArgs()
+	seederArgs.P2pConfig.Node.Transports = config.TransportConfig{
+		TCP: config.TCPProtocolConfig{
+			ListenAddress: libp2p.TestListenAddrWithIp4AndTcp,
+		},
+		QUICAddress:         "/ip4/127.0.0.1/udp/%d/quic",
+		WebSocketAddress:    "/ip4/127.0.0.1/tcp/%d/ws",
+		WebTransportAddress: "/ip4/127.0.0.1/udp/%d/quic-v1/webtransport",
+	}
+	seeder, err := libp2p.NewNetworkMessenger(seederArgs)
+	require.Nil(t, err)
+	messengers = append(messengers, seeder)
+
+	tcpPeerArgs := createBaseArgs()
+	tcpPeerArgs.P2pConfig.Node.Transports = config.TransportConfig{
+		TCP: config.TCPProtocolConfig{
+			ListenAddress: libp2p.TestListenAddrWithIp4AndTcp,
+		},
+	}
+	tcpPeer, err := libp2p.NewNetworkMessenger(tcpPeerArgs)
+	require.Nil(t, err)
+	addressToConnect := getAddressMatching(seeder.Addresses(), "/tcp/", "/ws/")
+	err = tcpPeer.ConnectToPeer(addressToConnect)
+	require.Nil(t, err)
+	messengers = append(messengers, tcpPeer)
+
+	quicPeerArgs := createBaseArgs()
+	quicPeerArgs.P2pConfig.Node.Transports = config.TransportConfig{
+		QUICAddress: "/ip4/127.0.0.1/udp/%d/quic",
```

Comment:
👋 Hi, I discovered this PR from this [tweet](https://twitter.com/SasuRobert/status/1667885367758839808)
It is awesome to see QUIC and WebTransport being added!

A drive-by suggestion I have is to use `quic-v1` for QUIC multiaddresses. `/quic/` is the [draft-29 version](https://docs.libp2p.io/concepts/transports/quic/#distinguishing-multiple-quic-versions-in-libp2p) of the address and go-libp2p is planning on removing support for it in the next release https://github.com/libp2p/go-libp2p/issues/2369
So it would make sense for any new QUIC deployment to use `quic-v1` right away

Also, you are already using `quic-v1` for your WebTransport addresses; using it in both places would enable inferring WebTransport listen addrs from observation (via Identify) of quic-v1 addrs https://github.com/libp2p/go-libp2p/pull/2251

Reply:
Thanks for the heads-up! 
Will change the mutiaddress to use `quick-v1` instead of `quick`

Reply:
done & pushed